python/pyarrow/_substrait.pyx

```diff
@@ -124,7 +129,10 @@ def run_query(plan, table_provider=None):
         function[CNamedTableProvider] c_named_table_provider
         CConversionOptions c_conversion_options
 
-    c_buf_plan = pyarrow_unwrap_buffer(plan)
+    if isinstance(plan, bytes):
+        c_buf_plan = pyarrow_unwrap_buffer(py_buffer(plan))
+    else:
+        c_buf_plan = pyarrow_unwrap_buffer(plan)
```

Comment:
```suggestion
    elif isinstance(plan, Buffer):
        c_buf_plan = pyarrow_unwrap_buffer(plan)
    else:
        raise TypeError(...)
```

Possible alternative to the fueed type. While that is a nice trick, it seems to give a confusing error message? ("No matching signature found" ?)